modules/libcom/test/aslibtest.c

```diff
@@ -111,6 +157,335 @@ static void testHostNames(void)
+"          - ops\n"
+"        methods:\n"
+"          - ca\n"
+"\n"
+"  # read write access for foo with a secure connection authenticated by Epics Org CA\n"
+"  - name: rw\n"
+"    links:\n"
+"      - INPA: ACC-CT{}Prmt:Remote-Sel\n"
+"      - INPB: ACC-CT{}Prmt:Remote-Sel\n"
+"    rules:\n"
+"      - level: 0\n"
+"        access: NONE\n"
+"      - level: 1\n"
+"        access: WRITE\n"
+"        trapwrite: true\n"
+//"        calc: VAL>=0\n"
+"        uags:\n"
+"          - foo\n"
+"        methods:\n"
+"          - x509\n"
+"        authorities:\n"
+"          - Epics Org CA\n"
+"\n"
+"  # RPC access for localhost user bar with a secure EPICS Org CA authenticated connection\n"
+"  - name: rwx\n"
+"    rules:\n"
+"      - level: 0\n"
+"        access: NONE\n"
+"      - level: 1\n"
+"        access: RPC\n"
+"        trapwrite: true\n"
+"        isTLS: true\n"
+"        uags:\n"
+"          - bar\n"
+"        hags:\n"
+"          - local\n"
+"        methods:\n"
+"          - x509\n"
+"          - ignored\n"
+"          - ignored_too\n"
+"        authorities:\n"
+"          - Epics Org CA\n"
+"          - ORNL Org CA\n"
+;
+
+static void testMethodAndAuthYaml(void)
+{
+    testDiag("testMethodAndAuthYaml()");
+    asCheckClientIP = 0;
+
+    testOk1(asInitMem(method_auth_config_yaml, NULL)==0);
+
+    asAsl = 0;
+    setHost("localhost");
+    setUser("foobar");
+    setMethod("ca");
+
+    testAccess("ro", 0);
+    testAccess("rw", 0);
+    testAccess("rwx", 0);
+
+    setUser("testing");
+
+    testAccess("ro", 1);
+    testAccess("rw", 0);
+    testAccess("rwx", 0);
+
+    setMethod("x509");
+
+    testAccess("ro", 0);
+    testAccess("rw", 0);
+    testAccess("rwx", 0);
+
+    setAuthority("Epics Org CA");
+    setIsTLS(1);
+
+    testAccess("ro", 0);
+    testAccess("rw", 3);
+    testAccess("rwx", 0);
+
+    setAuthority("Epics Org CA");
+
+    testAccess("ro", 0);
+    testAccess("rw", 3);
+
+    setUser("boss");
+    testAccess("rwx", 7);
+}
+
+static const char *expected_method_auth_config =
+  "UAG(bar) {boss}\n"
+  "UAG(foo) {testing}\n"
+  "UAG(ops) {geek}\n"
+  "ASG(DEFAULT) {\n"
+  "\tRULE(0,NONE,NOTRAPWRITE)\n"
+  "}\n"
+  "ASG(ro) {\n"
+  "\tRULE(0,NONE,NOTRAPWRITE)\n"
+  "\tRULE(1,READ,NOTRAPWRITE) {\n"
+  "\t\tUAG(foo,ops)\n"
+  "\t\tMETHOD(\"ca\")\n"
+  "\t}\n"
+  "}\n"
+  "ASG(rw) {\n"
+  "\tRULE(0,NONE,NOTRAPWRITE)\n"
+  "\tRULE(1,WRITE,TRAPWRITE) {\n"
+  "\t\tUAG(foo)\n"
+  "\t\tMETHOD(\"x509\")\n"
+  "\t\tAUTHORITY(\"Epics Org CA\")\n"
+  "\t}\n"
+  "}\n"
+  "ASG(rwx) {\n"
+  "\tRULE(0,NONE,NOTRAPWRITE)\n"
+  "\tRULE(1,RPC,NOTRAPWRITE) {\n"
+  "\t\tUAG(bar)\n"
+  "\t\tMETHOD(\"x509\",\"ignored\",\"ignored_too\")\n"
+  "\t\tAUTHORITY(\"Epics Org CA\",\"ignored\",\"ORNL Org CA\")\n"
+  "\t}\n"
+  "}\n";
+
+static void testDumpOutput(void)
+{
+    testDiag("testDumpOutput()");
+    testOk1(asInitMem(method_auth_config, NULL)==0);
+
+    char *buf = NULL;
+    size_t size = 0;
+    FILE *fp = open_memstream(&buf, &size);
```

Comment:
`open_memstream()` doesn't appear to be available on Windows, which is one of our supported platforms. We might consider dropping some of the older Windows compiler builds before this gets merged, but recent MSVC compilers must be able to build and run the tests.

Reply:
While I appreciate the avoidance of clutter, our tests are meant to run in an `O.<arch>/` sub-directory.  So it would be reasonable, and simpler, to write and read back a file.